### ISSUE TITLE
[a11y] Onboarding screen's X close button icon that has low color contrast

### ISSUE THREAD

**Issue by anna-yeddi** [mozilla-mobile/firefox-ios#27731]
## Steps to reproduce
1. Open Firefox for the first time
2. On the onboarding screens, locate a "X" button to close the onboarding and observe its appearance

### Expected behavior
The "X" close button uses the icon color that is easy to discern against the page background, with the color contrast being at least 3:1.

### Actual behavior
The "X" close button uses light gray color against the white background that gives low color contrast of 2.4:1 that makes this part of the UI inaccessible for users with low vision, users with cognitive disabilities, users with dyslexia, with colorblindness, users with screen glare, etc.

Contrast details (as checked from a screenshot):
- Foreground: #A8A5A8
- Background: #FFFFFF
- The contrast ratio is: 2.4:1, which fails for all WCAG 2.2 contrast criteria, incl:
    - 1.4.3 Contrast (Minimum) (AA) - Fail for large and regular text
    - 1.4.6 Contrast (Enhanced) (AAA) - Fail for large and regular text
    - 1.4.11 Non-text Contrast (AA) - Fail for UI components and graphical objects - this is the applicable criterion for this image button, because the functionality of the button is only provided via an image

### Device & build information
Screenshot captured from a user interview recording from July 1, 2025, thus details on the build and iOS version are not known
* Device: ? iPhone
* Build version: (current release)
* First seen version: ?
### Notes
Attachments: 

![X close button on the onboarding on the light theme with the Colour Contrast Analyzer showing Fail for 1.4.11 Non-text Contrast (AA) with 2.4:1 contrast ratio while 3:1 and more is needed](https://github.com/user-attachments/assets/3a96f4ae-e3de-45c2-b254-ade02ee4e9a7)
